.github/workflows/e2e-pull-request.yml

```diff
@@ -0,0 +1,57 @@
+name: E2E Tests - Pull Request
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - trunk
+  workflow_dispatch:
+
+env:
+  E2E_GH_TOKEN:          ${{ secrets.E2E_GH_TOKEN }}
+  WCP_DEV_TOOLS_REPO:    ${{ secrets.WCP_DEV_TOOLS_REPO }}
+  WCP_DEV_TOOLS_BRANCH:  'trunk'
+  WCP_SERVER_REPO:       ${{ secrets.WCP_SERVER_REPO }}
+  WC_SUBSCRIPTIONS_REPO: ${{ secrets.WC_SUBSCRIPTIONS_REPO }}
+  E2E_BLOG_ID:           ${{ secrets.E2E_BLOG_ID }}
+  E2E_BLOG_TOKEN:        ${{ secrets.E2E_BLOG_TOKEN }}
+  E2E_USER_TOKEN:        ${{ secrets.E2E_USER_TOKEN }}
+  WC_E2E_SCREENSHOTS:    1
+  E2E_SLACK_CHANNEL:     ${{ secrets.E2E_SLACK_CHANNEL }}
+  E2E_SLACK_TOKEN:       ${{ secrets.E2E_SLACK_TOKEN }}
+  E2E_USE_LOCAL_SERVER:  false
+  E2E_RESULT_FILEPATH:   'tests/e2e/results.json'
+
+
+jobs:
+  wcpay-e2e-tests:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast:     false
+      matrix:
+        woocommerce:   [ 'latest' ]
```

Comment:
We could probably remove these from the matrix and use `latest` directly in their places below for WP and WC versions since we are dealing with a single one.

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/4658/commits/0385ca5f10f61386ccd53a033c5ad6cfb9e02b79